prov/sockets/src/sock_ep_msg.c

```diff
@@ -1236,3 +1256,158 @@ int sock_msg_passive_ep(struct fid_fabric *fabric, struct fi_info *info,
 	return ret;
 }
 
+static void sock_ep_cm_check_msg_list(struct sock_ep_cm_head *cm_head)
+{
+	struct dlist_entry *entry;
+	struct sock_conn_req_handle *hreq;
+
+	fastlock_acquire(&cm_head->list_lock);
+	while (!dlist_empty(&cm_head->close_list)) {
+		entry = cm_head->close_list.next;
+		dlist_remove(entry);
+		hreq = container_of(entry, struct sock_conn_req_handle, entry);
+
+		sock_ep_cm_unmonitor_handle(hreq);
+		ofi_close_socket(hreq->sock_fd);
+		hreq->sock_fd = -1;
+
+		pthread_mutex_lock(&hreq->mutex_cond.mutex);
```

Comment:
fastlock converts to a spinlock.  We don't want to acquire a blocking mutex while holding a spinlock.

Reply:
Done